compiler/plc_xml/src/xml_parser.rs

```diff
@@ -26,21 +32,37 @@ mod pou;
 mod tests;
 mod variables;
 
-pub(crate) trait Parseable {
-    type Item;
-    fn visit(reader: &mut PeekableReader) -> Result<Self::Item, Error>;
+pub(crate) fn get_attributes(attributes: Attributes) -> Result<HashMap<String, String>, Error> {
+    attributes
+        .flatten()
+        .map(|it| Ok((it.key.try_to_string()?, it.value.try_to_string()?)))
+        .collect::<Result<HashMap<_, _>, Error>>()
+}
+
+pub(crate) trait Parseable
+where
+    Self: Sized,
+{
+    fn visit(reader: &mut Reader, tag: Option<BytesStart>) -> Result<Self, Error>;
 }
 
 pub(crate) fn visit(content: &str) -> Result<Project, Error> {
-    let mut reader = PeekableReader::new(content);
+    let mut reader = Reader::new(content);
+    reader.trim_text(true).expand_empty_elements(true);
+    let mut project = Project::default();
+
     loop {
-        match reader.peek()? {
-            Event::Start(tag) if tag.name().as_ref() == b"pou" => return Project::pou_entry(&mut reader),
-            Event::Start(tag) if tag.name().as_ref() == b"project" => return Project::visit(&mut reader),
-            Event::Eof => return Err(Error::UnexpectedEndOfFile(vec![b"pou"])),
-            _ => reader.consume()?,
+        match reader.read_event()? {
+            Event::Start(tag) if tag.name().as_ref() == b"pou" => {
+                project.pous.push(Pou::visit(&mut reader, Some(tag))?)
+            }
+            Event::Start(tag) if tag.name().as_ref() == b"project" || tag.name().as_ref() == b"pous" => {} //TODO: Handle some attributes
+            Event::End(tag) if tag.name().as_ref() == b"project" || tag.name().as_ref() == b"pous" => break,
```

Comment:
Is this TODO covered in [#977](https://github.com/PLC-lang/rusty/issues/977)?

Reply:
I think so yes, but it's not really required i'll remove it

Reply:
I added an actual todo error for this.